packages/editor/src/components/post-text-editor/index.js

```diff
@@ -45,68 +33,68 @@ export class PostTextEditor extends Component {
 	 *
 	 * @param {Event} event Change event.
 	 */
-	edit( event ) {
-		const value = event.target.value;
-		this.props.onChange( value );
-		this.setState( { value, isDirty: true } );
-	}
+	const edit = ( event ) => {
+		const val = event.target.value;
+
+		onChange( val );
+
+		setValue( val );
+		setIsDirty( true );
+	};
 
 	/**
 	 * Function called when the user has completed their edits, responsible for
 	 * ensuring that changes, if made, are surfaced to the onPersist prop
 	 * callback and resetting dirty state.
 	 */
-	stopEditing() {
-		if ( this.state.isDirty ) {
-			this.props.onPersist( this.state.value );
-			this.setState( { isDirty: false } );
+	const stopEditing = () => {
+		if ( isDirty ) {
+			onPersist( value );
+			setIsDirty( false );
 		}
-	}
+	};
 
-	render() {
-		const { value } = this.state;
-		const { instanceId } = this.props;
-		return (
-			<>
-				<VisuallyHidden
-					as="label"
-					htmlFor={ `post-content-${ instanceId }` }
-				>
-					{ __( 'Type text or HTML' ) }
-				</VisuallyHidden>
-				<Textarea
-					autoComplete="off"
-					dir="auto"
-					value={ value }
-					onChange={ this.edit }
-					onBlur={ this.stopEditing }
-					className="editor-post-text-editor"
-					id={ `post-content-${ instanceId }` }
-					placeholder={ __( 'Start writing with text or HTML' ) }
-				/>
-			</>
-		);
-	}
+	return (
+		<>
+			<VisuallyHidden
+				as="label"
+				htmlFor={ `post-content-${ instanceId }` }
+			>
+				{ __( 'Type text or HTML' ) }
+			</VisuallyHidden>
+			<Textarea
+				autoComplete="off"
+				dir="auto"
+				value={ value }
+				onChange={ edit }
+				onBlur={ stopEditing }
+				className="editor-post-text-editor"
+				id={ `post-content-${ instanceId }` }
+				placeholder={ __( 'Start writing with text or HTML' ) }
+			/>
+		</>
+	);
 }
 
-export default compose( [
-	withSelect( ( select ) => {
-		const { getEditedPostContent } = select( 'core/editor' );
-		return {
-			value: getEditedPostContent(),
-		};
-	} ),
-	withDispatch( ( dispatch ) => {
-		const { editPost, resetEditorBlocks } = dispatch( 'core/editor' );
-		return {
-			onChange( content ) {
+export default function ComposedPostTextEditor() {
+	const value = useSelect(
```

Comment:
You actually don't have to create another function ( `ComposedPostTextEditor` ) to replace the previous composition. You can use all the needed hooks, such as `useDispatch` etc, in the same function component above.

What this results to is that `value`, `onChange` and `onPersist` will not be extracted from `props`.

You should also check how to properly refactor `getDerivedStateFromProps`:  https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops.

You can also check some of the merged PRs here: https://github.com/WordPress/gutenberg/issues/22890, if you need some guidance.

Reply:
> You actually don't have to create another function ( `ComposedPostTextEditor` ) to replace the previous composition. You can use all the needed hooks, such as `useDispatch` etc, in the same function component above.
> 
> What this results to is that `value`, `onChange` and `onPersist` will not be extracted from `props`.

I was thinking about doing this but this would result in losing the ability to call the PostTextEditor with a value in the props. Is this function no longer required?

Reply:
> Is this function no longer required?

If you mean the composition, then you're right. With hooks, you can extract the values inside the function component.

For example the `value` prop ( previously added to `PostTextEditor` by composition with `withSelect` higher order component ) would be inside like:
```js
const value = useSelect( ( select ) => select( 'core/editor' ).getEditedPostContent() );
```

You can check here more info: https://developer.wordpress.org/block-editor/packages/packages-data/#withSelect, and be sure to check the above link for `getDerivedStateFromProps`

> losing the ability to call the PostTextEditor with a value in the props

You can see value is extracted from the store with `getEditedPostContent` and is not passed directly. It's not expected to be passed. It ends up in props now due to the composition.

Reply:
Thanks. I meant if the ability to pass the value from the parent is also no longer needed. I have done the refactoring and is rewriting the test cases.